internal/common/rules/basicruleevaluator_test.go

```diff
@@ -0,0 +1,53 @@
+package rules
+
+import (
+	"gonum.org/v1/gonum/mat"
+	"testing"
+)
+
+// TestBasicRuleEvaluatorPositive Checks whether rule we expect to evaluate as true actually evaluates as such
+func TestBasicRuleEvaluatorPositive(t *testing.T) {
+	result, err := BasicBooleanRuleEvaluator("Kinda Complicated Rule")
+	if !result {
+		t.Errorf("Rule evaluation came as false, when it was expected to be true, potential error with value '%v'", err)
+	}
+}
+
+// TestBasicRuleEvaluatorNegative Checks whether rule we expect to evaluate as false actually evaluates as such
+func TestBasicRuleEvaluatorNegative(t *testing.T) {
+	registerTestRule()
+	result, err := BasicBooleanRuleEvaluator("Kinda Test Rule")
+	if result || err != nil {
+		t.Errorf("Rule evaluation came as true, when it was expected to be false, potential error with value '%v'", err)
+	}
+}
+
+func TestBasicRealValuedRuleEvaluator(t *testing.T) {
+	registerNewRealValuedRule(t)
+	result, value, _ := BasicRealValuedRuleEvaluator("Real Test rule")
+	if !result && value != 2.0 {
+		t.Errorf("Real values rule evaluation error, expected true got '%v', value expected '2' got '%v'", result, value)
+	}
+}
+
+func registerNewRealValuedRule(t *testing.T) {
+	//A very contrived rule//
+	name := "Real Test rule"
+	reqVar := []string{
+		"number_of_islands_contributing_to_common_pool",
+		"number_of_failed_forages",
+		"number_of_broken_agreements",
+		"max_severity_of_sanctions",
+	}
+
+	v := []float64{1, 0, 0, 0, -4, 0, -1, -1, 0, 2, 0, 0, 0, 0, 2, 0, 0, 1, 0, -1}
```

Comment:
Is there a reason why you dont write it up as a matrix straight away?
```
matrix := [][]float64
  {1, 0, 0, 0},
  {-4, 0,-1,-1},
  {0, 2, 0, 0},
  {0, 0, 2, 0} ,
  {0, 1, 0, -1}}

Reply:
Hmmm  I did it this way because of how we feed it into gonums newDense function, if we can feed it as a multi-dimensional that would be nicer. Will investigate later